docs/dev/compilation-time-analysis.md

```diff
@@ -0,0 +1,253 @@
+ScyllaDB's long build time has been a problem almost since its inception -
+our issue #1 - https://github.com/scylladb/scylladb/issues/1 - is about it.
+To reduce Scylla's compilation time, it's important to understand what is
+causing it.
+
+In the old days of of C compilers, the compilation time was mostly
+proportional to the amount of source code. The only way to reduce compilation
+time without deleting source code was to reduce the amount of unnecessary
+header-file inclusion. However, in modern template-heavy C++ code, such
+as Seastar and ScyllaDB, not all header files slow down the build by
+the same amount. Template instantiation and code generation takes orders
+of magnitude more time than just parsing - and it's possible that one
+header file adds a millisecond to the build time, while another header
+file of the same length adds a full second.
+
+Clang has a compile-time option, "-ftime-trace", for tracing the
+compilation time of a single source file, to find which parts of the
+compilation took most time. This document is about how to use this
+option together with a tool `ClangBuildAnalyzer` - which aggregates the
+`-ftime-trace` logs from all the source files to decide which header files
+and templates are slowing down build the most.
+
+# Installing ClangBuildAnalyzer
+
+Get ClangBuildAnalyzer:
+```
+git clone https://github.com/aras-p/ClangBuildAnalyzer
+```
+Build it with `make -f projects/make/Makefile`, and install the resulting
+`build/ClangBuildAnalyzer`.
+
+# Getting started with ClangBuildAnalyzer
+In your Scylla working directory, start with a fresh "build" directory
+to avoid measurement of already-deleted source files or irrelevant build
+modes:
+```
+    rm -r build
+```
+Next, rebuild build.ninja to add `-ftime-trace` to the compilation commands.
+Run your favorite `configure.py` command line, adding `--cflags=-ftime-trace`.
+For example:
+
+```
+    ./configure.py --disable-dpdk --cflags=-ftime-trace
+```
+
+Now build the main `scylla` executable, in one build mode. For example,
+```
+    ninja  build/dev/scylla
+```
+
+This build will create in `build/dev`, for each source file compiled, a
+**separate** JSON file with the compile-time analysis of it. For example,
+for `cdc/generation.cc`, a file `build/dev/cdc/generation.json` will be
+created.
+
+To make sense of these almost-unreadable files, and to combine the
```

Comment:
You should mention that the JSON files are in chrome trace format, which can be viewed in multiple tools, notably Google's trace viewer: https://ui.perfetto.dev/

![image](https://github.com/scylladb/scylladb/assets/45052050/b2331a78-bf46-41fe-8f14-9ad330536d2a)